ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -250,8 +276,8 @@
       }
       // only need to issue an update statement if the row was not already marked as deleted.
```

Comment:
What happens to the locked record if the update is not made?  In either case, when does it get unlocked (not sure if this whole thing is inside a transaction)?

Reply:
Yep this whole thing is inside a transaction. In this case, the row is unlocked when the transaction is finished (`executeTransactionAsync` calls transaction.commit() or rollback())